sdk/search/azure-search-perf/src/main/java/com/azure/search/perf/core/ServiceTest.java

```diff
@@ -0,0 +1,111 @@
+
+package com.azure.search.perf.core;
+
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.perf.test.core.PerfStressTest;
+import com.azure.search.documents.SearchAsyncClient;
+import com.azure.search.documents.SearchClient;
+import com.azure.search.documents.indexes.SearchIndexAsyncClient;
+import com.azure.search.documents.indexes.SearchIndexClientBuilder;
+import com.azure.search.documents.indexes.models.IndexDocumentsBatch;
+import com.azure.search.documents.indexes.models.SearchIndex;
+import com.azure.search.documents.indexes.models.SearchSuggester;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Base class for Azure Search performance tests.
+ */
+public abstract class ServiceTest<TOptions extends PerfStressOptions> extends PerfStressTest<TOptions> {
+    private static final String CONFIGURATION_ERROR = "Configuration %s must be set in either environment variables "
+        + "or system properties.%n";
+    private static final String ALLOWED_INDEX_CHARACTERS = "abcdefghijklmnopqrstuvwxyz0123456789";
+    private static final int INDEX_NAME_LENGTH = 24;
+
+    protected static final String SUGGESTER_NAME = "sg";
+
+    protected final SearchClient searchClient;
+    protected final SearchAsyncClient searchAsyncClient;
+
+    private final SearchIndexAsyncClient searchIndexAsyncClient;
+    private final String indexName;
+
+    public ServiceTest(TOptions options) {
+        super(options);
+
+        String searchEndpoint = Configuration.getGlobalConfiguration().get("SEARCH_ENDPOINT");
+        if (CoreUtils.isNullOrEmpty(searchEndpoint)) {
+            System.out.printf(CONFIGURATION_ERROR, "SEARCH_ENDPOINT");
+            System.exit(1);
+        }
+
+        String searchApiKey = Configuration.getGlobalConfiguration().get("SEARCH_API_KEY");
+        if (CoreUtils.isNullOrEmpty(searchApiKey)) {
+            System.out.printf(CONFIGURATION_ERROR, "SEARCH_API_KEY");
+            System.exit(1);
+        }
+
+        SearchIndexClientBuilder builder = new SearchIndexClientBuilder()
+            .endpoint(searchEndpoint)
+            .credential(new AzureKeyCredential(searchApiKey))
+            .httpClient(new NettyAsyncHttpClientBuilder().build());
+
+        this.searchIndexAsyncClient = builder.buildAsyncClient();
+
+        Random random = new Random();
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < INDEX_NAME_LENGTH; i++) {
+            stringBuilder.append(ALLOWED_INDEX_CHARACTERS.charAt(random.nextInt(ALLOWED_INDEX_CHARACTERS.length())));
+        }
+
+        this.indexName = stringBuilder.toString();
```

Comment:
The indexName generation logic can be optionally replaced with this.

```java
 this.indexName  =  random.ints(0, ALLOWED_INDEX_CHARACTERS.length())
            .limit(INDEX_NAME_LENGTH)
            .collect(StringBuilder::new, ((stringBuilder, value) -> stringBuilder.append(ALLOWED_INDEX_CHARACTERS.charAt(value))), StringBuilder::append)
            .toString();
```

Reply:
Will do a slight variant on this, using `random.ints(long streamSize, int lowerBound, int upperBound)`.